src/pages/PoolDetails/PoolDetailsHeader.tsx

```diff
@@ -0,0 +1,209 @@
+        <StyledLink to={origin}>
+          <ThemedText.BodySecondary>
+            <Trans>Pool</Trans>
+          </ThemedText.BodySecondary>
+        </StyledLink>
+        <ThemedText.BodySecondary>&nbsp;{'>'}&nbsp;</ThemedText.BodySecondary>
+        <ThemedText.BodyPrimary>
+          {token0?.symbol} / {token1?.symbol} ({shortenAddress(poolAddress)})
+        </ThemedText.BodyPrimary>
+      </Row>
+      <Row gap="18px">
+        <Row gap="8px" width="max-content">
+          {chainId && (
+            <DoubleCurrencyAndChainLogo data-testid="double-token-logo" chainId={chainId} currencies={currencies} />
+          )}
+          <ThemedText.HeadlineSmall>
+            {token0?.symbol} / {token1?.symbol}
+          </ThemedText.HeadlineSmall>
+        </Row>
+        {feeTier && <FeeTier>{feeTier / 10000}%</FeeTier>}
+        <ToggleReverseArrows data-testid="toggle-tokens-reverse-arrows" onClick={toggleReversed} />
+      </Row>
+    </HeaderColumn>
+  )
+}
+
+const StyledLogoParentContainer = styled.div`
+  position: relative;
+  top: 0;
+  left: 0;
+`
+
+function DoubleCurrencyAndChainLogo({
+  chainId,
+  currencies,
+}: {
+  chainId: number
+  currencies: Array<Currency | undefined>
+}) {
+  return (
+    <StyledLogoParentContainer>
+      <DoubleCurrencyLogo chainId={chainId} currencies={currencies} />
+      <SquareL2Logo chainId={chainId} />
+    </StyledLogoParentContainer>
+  )
+}
+
+const L2LogoContainer = styled.div<{ hasSquareLogo?: boolean }>`
+  background-color: ${({ theme, hasSquareLogo }) => (hasSquareLogo ? theme.surface2 : theme.neutral1)};
+  border-radius: 2px;
+  height: 12px;
+  left: 60%;
+  position: absolute;
+  top: 60%;
+  outline: 2px solid ${({ theme }) => theme.surface1};
+  width: 12px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`
+
+const StyledChainLogo = styled.img`
+  height: 12px;
+  width: 12px;
+`
+
+const SquareChainLogo = styled.img`
+  height: 100%;
+  width: 100%;
+`
+
+function SquareL2Logo({ chainId }: { chainId: ChainId }) {
+  if (chainId === ChainId.MAINNET) return null
+  const { squareLogoUrl, logoUrl } = getChainInfo(chainId)
+
+  const chainLogo = squareLogoUrl ?? logoUrl
+
+  return (
+    <L2LogoContainer hasSquareLogo={!!squareLogoUrl}>
+      {squareLogoUrl ? (
+        <SquareChainLogo src={chainLogo} alt="chainLogo" />
+      ) : (
+        <StyledChainLogo src={chainLogo} alt="chainLogo" />
+      )}
+    </L2LogoContainer>
+  )
+}
+
+function DoubleCurrencyLogo({ chainId, currencies }: { chainId: number; currencies: Array<Currency | undefined> }) {
+  const [src, nextSrc] = useTokenLogoSource(currencies?.[0]?.wrapped.address, chainId, currencies?.[0]?.isNative)
+  const [src2, nextSrc2] = useTokenLogoSource(currencies?.[1]?.wrapped.address, chainId, currencies?.[1]?.isNative)
+
+  return <DoubleLogo logo1={src} onError1={nextSrc} logo2={src2} onError2={nextSrc2} />
+}
+
+const DoubleLogoContainer = styled.div`
+  display: flex;
+  flex-direction: row;
```

Comment:
```suggestion

```

Reply:
nit: row is the default (not important to remove--sometimes I like being explicit about this)